skbio/stats/composition.py

```diff
@@ -533,12 +579,20 @@ def clr_inv(mat):
 
     """
     # for numerical stability (aka softmax trick)
-    mat = np.atleast_2d(mat)
-    emat = np.exp(mat - mat.max(axis=-1, keepdims=True))
-    return closure(emat)
-
+    xp = aac.array_namespace(mat)
+    
+    # ensure last dimension is not singleton
+    if xp.shape(mat)[-1] == 1:
+        for i in range(xp.shape(mat),0,-1):
+            if xp.shape(mat)[i] == 1:
+                mat = xp.squeeze(mat, axis=i)
+            else:
+                break
+    
+    gm = xp.mean(mat, axis=-1, keepdims=True)
+    return xp.exp(mat + gm)
 
-def ilr(mat, basis=None, check=True):
+def ilr(mat:Array, basis:Array|None=None, /, check:bool=True)-> Array:
```

Comment:
scikit-bio still supports Python 3.9, which doesn't allow `|` in type hinting. So `Array|None` needs to be `Optional[Array]` instead. Likewise, `int|Array` below needs to be `Union[int, Array]`.

Reply:
Let's not forget to remove the `/` here